test/box-py/iproto.test.py

```diff
@@ -45,12 +45,31 @@
             not k.startswith("IPROTO_SQL_INFO_") and type(v) == int:
         key_names[v] = k
 
+def deep_convert_dict_to_list(obj):
+    ret = []
+    if isinstance(obj, dict):
+        for k, v in sorted(obj.items()):
+            if hasattr(v, '__getitem__'):
+                ret.append((k, deep_convert_dict_to_list(v)))
+            else:
+                ret.append((k, v))
+        return ret
+    if isinstance(obj, list):
+        for v in obj:
+            if hasattr(v, '__getitem__'):
+                ret.append(deep_convert_dict_to_list(v))
+            else:
+                ret.append(v)
+        return ret
+
+    return obj
+
 def repr_dict(todump):
     d = {}
     for (k, v) in todump.items():
         k_name = key_names.get(k, k)
         d[k_name] = v
-    return repr(d)
+    return repr(deep_convert_dict_to_list(d))
```

Comment:
The problem does not appear on my machine, however, I see, it is possible in theory. Anyway: it seems, the following patch is sufficient:

```diff
diff --git a/test/box-py/iproto.test.py b/test/box-py/iproto.test.py
index 10ea24d44..83fc7f849 100644
--- a/test/box-py/iproto.test.py
+++ b/test/box-py/iproto.test.py
@@ -50,7 +50,7 @@ def repr_dict(todump):
     for (k, v) in todump.items():
         k_name = key_names.get(k, k)
         d[k_name] = v
-    return repr(d)
+    return repr(sorted(d.items()))
 
 def test(header, body):
     # Connect and authenticate
```

Reply:
how original problem looks like:
```python
[009] box-py/iproto.test.py                                           [ fail ]
[009] 
[009] Test failed! Result content mismatch:
[009] --- box-py/iproto.result	Sat Mar 13 17:46:54 2021
[009] +++ /rw_bins/test/var/rejects/box-py/iproto.reject	Sat Mar 13 17:50:51 2021
[009] @@ -29,7 +29,7 @@
[009]  IPROTO_UPDATE
[009]  query {'IPROTO_CODE': 4} {'IPROTO_SPACE_ID': 280}
[009]  True
[009] -query {'IPROTO_CODE': 4} {'IPROTO_SPACE_ID': 280, 'IPROTO_KEY': (1,)}
[009] +query {'IPROTO_CODE': 4} {'IPROTO_KEY': (1,), 'IPROTO_SPACE_ID': 280}
[009]  True
[009]  
[009]  
[009] 
[009] Last 15 lines of Tarantool Log file [Instance "box"][/rw_bins/test/var/009_box-py/box.log]:
```

1. https://github.com/tarantool/tarantool/pull/5751/checks?check_run_id=2103097418
2. https://github.com/tarantool/tarantool/pull/5751/checks?check_run_id=2103097497